include/revng/Model/Register.h

```diff
@@ -681,6 +682,242 @@ fromCSVName(llvm::StringRef Name, model::Architecture::Values Architecture) {
+  case x27_aarch64:
+  case x28_aarch64:
+  case x29_aarch64:
+  case lr_aarch64:
+  case sp_aarch64:
+  case v0_mips:
+  case v1_mips:
+  case a0_mips:
+  case a1_mips:
+  case a2_mips:
+  case a3_mips:
+  case s0_mips:
+  case s1_mips:
+  case s2_mips:
+  case s3_mips:
+  case s4_mips:
+  case s5_mips:
+  case s6_mips:
+  case s7_mips:
+  case t0_mips:
+  case t1_mips:
+  case t2_mips:
+  case t3_mips:
+  case t4_mips:
+  case t5_mips:
+  case t6_mips:
+  case t7_mips:
+  case t8_mips:
+  case t9_mips:
+  case gp_mips:
+  case sp_mips:
+  case fp_mips:
+  case ra_mips:
+  case r0_systemz:
+  case r1_systemz:
+  case r2_systemz:
+  case r3_systemz:
+  case r4_systemz:
+  case r5_systemz:
+  case r6_systemz:
+  case r7_systemz:
+  case r8_systemz:
+  case r9_systemz:
+  case r10_systemz:
+  case r11_systemz:
+  case r12_systemz:
+  case r13_systemz:
+  case r14_systemz:
+  case r15_systemz:
+    return model::PrimitiveTypeKind::PointerOrNumber;
+
+  case xmm0_x86:
+  case xmm1_x86:
+  case xmm2_x86:
+  case xmm3_x86:
+  case xmm4_x86:
+  case xmm5_x86:
+  case xmm6_x86:
+  case xmm7_x86:
+  case xmm0_x86_64:
+  case xmm1_x86_64:
+  case xmm2_x86_64:
+  case xmm3_x86_64:
+  case xmm4_x86_64:
+  case xmm5_x86_64:
+  case xmm6_x86_64:
+  case xmm7_x86_64:
+  case q0_arm:
+  case q1_arm:
+  case q2_arm:
+  case q3_arm:
+  case q4_arm:
+  case q5_arm:
+  case q6_arm:
+  case q7_arm:
+  case v0_aarch64:
+  case v1_aarch64:
+  case v2_aarch64:
+  case v3_aarch64:
+  case v4_aarch64:
+  case v5_aarch64:
+  case v6_aarch64:
+  case v7_aarch64:
+  case v8_aarch64:
+  case v9_aarch64:
+  case v10_aarch64:
+  case v11_aarch64:
+  case v12_aarch64:
+  case v13_aarch64:
+  case v14_aarch64:
+  case v15_aarch64:
+  case v16_aarch64:
+  case v17_aarch64:
+  case v18_aarch64:
+  case v19_aarch64:
+  case v20_aarch64:
+  case v21_aarch64:
+  case v22_aarch64:
+  case v23_aarch64:
+  case v24_aarch64:
+  case v25_aarch64:
+  case v26_aarch64:
+  case v27_aarch64:
+  case v28_aarch64:
+  case v29_aarch64:
+  case v30_aarch64:
+  case v31_aarch64:
+  case f0_mips:
+  case f1_mips:
+  case f2_mips:
+  case f3_mips:
+  case f4_mips:
+  case f5_mips:
+  case f6_mips:
+  case f7_mips:
+  case f8_mips:
+  case f9_mips:
+  case f10_mips:
+  case f11_mips:
+  case f12_mips:
+  case f13_mips:
+  case f14_mips:
+  case f15_mips:
+  case f16_mips:
+  case f17_mips:
+  case f18_mips:
+  case f19_mips:
+  case f20_mips:
+  case f21_mips:
+  case f22_mips:
+  case f23_mips:
+  case f24_mips:
+  case f25_mips:
+  case f26_mips:
+  case f27_mips:
+  case f28_mips:
+  case f29_mips:
+  case f30_mips:
+  case f31_mips:
+  case f0_systemz:
+  case f1_systemz:
+  case f2_systemz:
+  case f3_systemz:
+  case f4_systemz:
+  case f5_systemz:
+  case f6_systemz:
+  case f7_systemz:
+  case f8_systemz:
+  case f9_systemz:
+  case f10_systemz:
+  case f11_systemz:
+  case f12_systemz:
+  case f13_systemz:
+  case f14_systemz:
+  case f15_systemz:
+    return model::PrimitiveTypeKind::Float;
```

Comment:
NOTETOSELF: cry

Reply:
I thought for a really long time for a better way, sadly this is what it came down to 😢